locals.tf

```diff
@@ -1,4 +1,12 @@
 locals {
+  # Flatten API operations into a single map for resource creation
+  api_operations = merge([
+    for api_key, api in var.apis : {
+      for op_key, op in api.operations : "${api_key}-${op_key}" => merge(op, {
```

Comment:
nit: 
let's call these `operation_key` and `operation`. Terraform variable naming conventions prefer longer names to shortened ones.